backend/src/modules/doubles-tournament/core/domain/QueuedMatch.ts

```diff
@@ -0,0 +1,70 @@
+import { TeamId } from './TeamId';
+import { TournamentId } from './TournamentId';
+import { DoublesTournament } from './DoublesTournament';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+import { MatchesQueue } from './MatchesQueue';
+import { MatchWasQueued } from './event/MatchWasQueued';
+
+export class QueuedMatch {
+  readonly matchNumber: number;
+  readonly team1Id: TeamId;
+  readonly team2Id: TeamId;
+
+  constructor(props: { matchNumber: number; team1Id: TeamId; team2Id: TeamId }) {
+    this.matchNumber = props.matchNumber;
+    this.team1Id = props.team1Id;
+    this.team2Id = props.team2Id;
+  }
+}
+
+export function pushMatchToQueue(
+  tournament: DoublesTournament | undefined,
+  queue: MatchesQueue | undefined,
+  command: {
+    tournamentId: TournamentId;
+    matchNumber: number;
+    team1Id: TeamId;
+    team2Id: TeamId;
+  },
+  currentTime: Date,
+): DomainCommandResult<MatchesQueue> {
+  if (tournament === undefined) {
+    throw new Error("This tournament doesn't exists.");
+  }
+  if (command.matchNumber <= 0) {
```

Comment:
Fajnie z MatchNumbera tez zrobic value object, bo np. wiemy, ze to nie jest kazdy numer, ze np. minimum to jest 1 (czy 0? - nie wiem ja kbędziemy numerowac, ale chyba od 1). 
Po to jest ValueObject, żeby nie było tych sprawdzeń. W domain nie ma czegoś takiego co nie istnieje. Czyli jak coś jest matchNumber, to nim jest - dlatego robisz typ i domain już m same poprawne wartości.

Podsumowujac: W domenie nie moze istnieć np. człowiek, który nie ma imienia. 
Czy wlasnie numer meczu, który nie może się zdarzyć.